table_one.py

```
@@ -343,11 +352,17 @@ def calculate_p_categorical(df, col, group_col):
+        # === INTEGRATION: Connection Handler ===
+        _chi2, p_chi2, _dof, ex = CONNECTION_HANDLER.retry_with_backoff(
+            lambda: stats.chi2_contingency(tab)
+        )
         
         if is_2x2 and ex.min() < 5:
             try:
-                _oddsr, p = stats.fisher_exact(tab)
+                _oddsr, p = CONNECTION_HANDLER.retry_with_backoff(
+                    lambda: stats.fisher_exact(tab)
+                )
                 return p, "Fisher's Exact"
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)